### DISCUSSION TITLE
How to do get basic `corelcr.so` symbols displayed with `gdb`?

### DISCUSSION THREAD

**Discussion by vadimkantorov** [dotnet/runtime#106258] [Q&A] [answered]
I'm encountering a SEGFAULT in an app hosting .NET Core 7.0's corelcr.so (via PythonNet). Is it possible to get symbols for gdb somewhere? And how to set up gdb to show at least basic function info? (even without SOS niceties)

![image](https://github.com/user-attachments/assets/a084179f-f8c5-4156-889f-86007cbf1a52)

Thanks!

**Comment by EgorBo** [comment, marked as answer]
https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-symbol

**Reply by vadimkantorov** [reply]
This page does not mention `gdb` at all. Probably this is simple and clear for experienced users, but would you have an advice on how to set up `gdb` to use Microsoft public symbol server? Thanks!

**Reply by EgorBo** [reply]
The tool can download symbols for libcoreclr.so (you just do something like `dotnet-symbol --symbols path/to/libcoreclr.so`, then it's literally the same workflow as for any other native binary. I think gdb should pick it up automatically if they're located next to binary.

More info: https://github.com/dotnet/symstore/blob/main/src/dotnet-symbol/README.md, also, gdb docs

**Reply by vadimkantorov** [reply]
Thanks! I'll try that! Also, `symstore` repo is for some reason recently "archived", hmm...

**Reply by AustinWise** [reply]
New location of the `dotnet-symbol` docs mentioned above: https://github.com/dotnet/diagnostics/blob/main/src/Tools/dotnet-symbol/README.md

**Reply by vadimkantorov** [reply]
Thanks a bunch! `dotnet-symbol` worked! I get now a symbol'ed stack trace
